sdk/core/allocator/main.cc

```diff
@@ -1048,26 +1052,23 @@ __cheriot_minimum_stack(0x230) void *heap_allocate_array(
   uint32_t            flags)
 {
 	STACK_CHECK(0x230);
-	if (!check_timeout_pointer(timeout))
+	if (!check_timeout_pointer(timeout) ||
+	    !check_pointer<PermissionSet{Permission::Load, Permission::Store}>(
```

Comment:
`check_timeout_pointer` already checks permissions so the second half of this condition is redundant. Looks like this came from `futex_timed_wait` (perhaps copied by AI?). We should fix that in another PR.